Emby.Server.Implementations/Channels/ChannelPostScanTask.cs

```diff
@@ -35,64 +35,52 @@ public Task Run(IProgress<double> progress, CancellationToken cancellationToken)
         public static string GetUserDistinctValue(User user)
         {
             var channels = user.Policy.EnabledChannels
-                .OrderBy(i => i)
-                .ToList();
+                .OrderBy(i => i);
 
-            return string.Join("|", channels.ToArray());
+            return string.Join("|", channels);
         }
 
         private void CleanDatabase(CancellationToken cancellationToken)
         {
             var installedChannelIds = ((ChannelManager)_channelManager).GetInstalledChannelIds();
 
-            var databaseIds = _libraryManager.GetItemIds(new InternalItemsQuery
+            var databaseIds = _libraryManager.GetItemList(new InternalItemsQuery
             {
-                IncludeItemTypes = new[] { typeof(Channel).Name }
+                IncludeItemTypes = new[] { typeof(Channel).Name },
+                ExcludeItemIds = installedChannelIds.ToArray()
```

Comment:
is this `.ToArray()` needed here?

Reply:
Trust me, I wouldn't want to allocate too much :p

Reply:
I believe you, can you enlighten me then why it's needed here?

Reply:
@JustAMan The `GetInstalledChannelIds`function returns an `IEnumerable<Guid>` instead of an `Guid[]`
https://github.com/jellyfin/jellyfin/blob/8487319374fa86601a6fcfdae64ee75dc51ed357/Emby.Server.Implementations/Channels/ChannelManager.cs#L132

Reply:
Well, so what? I mean isn't it enough to do its job later on? Are you saying the code needs to traverse the result multiple times? I hoped it would end in one `WHERE` clause, and for joining stuff you don't need it to be an array, right?